packages/shared/src/events/generic-event-emitter.ts

```diff
@@ -9,7 +9,7 @@ import { ProviderEvents } from './events';
 export abstract class GenericEventEmitter<AdditionalContext extends Record<string, unknown> = Record<string, unknown>>
   implements ManageLogger<GenericEventEmitter<AdditionalContext>>
 {
-  protected abstract readonly eventEmitter: NodeJS.EventEmitter;
```

Comment:
This could create an issue in some TS consumers which perform type-checking in libs when they compile/bundle.

Fixed by adding the private type polyfill below.